packages/python/libsql-client/libsql_client/http_driver.py

```diff
@@ -1,26 +1,56 @@
 from typing import Any, Dict, List, Optional
+from urllib.parse import urljoin
 import aiohttp
 import base64
 import json
 
 from .driver import _Driver, _RawStmt
-from .errors import ClientResponseError, ClientHttpError
+from .errors import ClientResponseError, ClientHttpError, ClientError
 from .result import ResultSet, Row, Value
 
 class _HttpDriver(_Driver):
     _session: aiohttp.ClientSession
     _url: str
+    _version: Optional[tuple[int, int, int]]
 
     def __init__(self, url: str) -> None:
         self._session = aiohttp.ClientSession()
         self._url = url
+        self._version = None
+
+    async def session(self):
```

Comment:
In Python, the convention is to prefix private methods with `_`. You can resolve the clash with `_session` by renaming it to `_get_session()` or, even better, to `_get_version()` (and make it return the version, not the session).

Reply:
I already have a `_get_version` method, the goal here is to lazy retrieve the version number when an HTTP call is made.

Reply:
so one should be using the `get_sessions()` method instead of the `_session` field

Reply:
I understand your intent, but there are two issues:
- Private methods should be prefixed with an underscore
- The functionality of this method has nothing to do with the session, it deals with the server version, so it should be named accordingly and it should return an appropriate value

Reply:
private method addressed

what do you suggest? the goal is to ensure that the session is only accessed with an adequately initialized driver. would moving the logic that sets the version to methods like `_ensure_client_init` make sense to you?

Reply:
I would suggest renaming the method to `_get_cached_version()` or something similar and let it return the version as `Tuple[int, int, int]`. You don't need to call this method every time you need to use the session (e.g. in `_get_version()`!), you need to call this method when you need to _know the server version_. The type system would nudge you: `self._version` is `Optional[Tuple]`, while the result from `_ensure_version()` would be just a `Tuple`.

(The "cached" in the name would also serve as a hint that the version is cached and may be invalid. This is the reason why HTTP APIs typically don't rely on clients querying the server version, but they put the burden of compatibility on the servers: the client specifies the version it expects.

A typical solution is that HTTP servers expose versioned paths, and the clients target the specific version that they expect. For an example, see `iku-turso-api`, which exposes `/v1` and `/v2` versions.)

Reply:
yeah, we're not going down the version API road just now, or we'll be on v40 by the end of the month.
The goal here is to allow a new client to work with an old DB
I'm not calling `_get_version()` every time, but only once per session
I don't care where the check is done, but it has to be done once, preferably somewhere where the control flow converges

Reply:
This check can be removed when users are migrated. 

We could also crash and return that the version of the server is incompatible with the client, and that it should be updated

Reply:
You have a clear idea of what you want, so, what do we do now to let old db work with new client? Just tell me, I'll do it

Reply:
If we want to keep compatiblity with old server, why don't we just keep using the "/" path in the client? This is what the PR would do anyway: we read the version from the server, but we don't use it.

In other words, if we care about compatibility, we should first upgrade the servers to support the new API in addition to the old API, then upgrade the clients to use the new API unconditionally.

If we don't care about compatibility, just switch both the server and the client to the new API.

Reply:
When we know what will be the version that contains the `/queries` route, we can check the version in the client. For now, we're not certain what will be the version number when we release `/queries`.

The proposed PR for `/queries` handled redirection of `/` to `/queries`, but what is not possible is to make old servers redirect `/queries` to `/`, and this is what we're trying to achieve here. The part you are describing is already covered.

Reply:
I just want to know how you suggest we handle the case where an old server (supports *only* `/`) is used with a new client (queries `/queries`)

Reply:
I suggest that we introduce new clients (using `/queries`) only after we upgrade our servers to handle both `/` and `/queries`.

Reply:
Server are not upgraded, that's the thing

Reply:
only some are, manually

Reply:
Hmm, maybe that's the root of the problem that should be fixed? :D

If we need to support old clients, the server must handle `/`, and if we need to support old servers, the clients must handle `/`, so this would mean that we are stuck with `/` anyway and we don't need to migrate clients to `/queries` :shrug:

However, I understand that we are laboring under constraints not under our control, so if we absolutely need to change the client behavior based on server version, I still think that we should refactor this function to return the version instead of the session, and use it only when we need the version.